data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/PeerForwarderProvider.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.peerforwarder;
+
+public class PeerForwarderProvider {
+    private final PeerForwarder peerForwarder;
+
+    public PeerForwarderProvider(final PeerForwarder peerForwarder) {
+        this.peerForwarder = peerForwarder;
+    }
+
+    public PeerForwarder register() {
```

Comment:
Another idea I considered is that `register` could take in the `identificationKeys` and `pluginId`. This way, each client gets its own `PeerForwarder` instance.

Reply:
Also, include `pipelineName` here.